build-site.sh

```diff
@@ -34,26 +34,29 @@ if [ ! -f "/srv/source/Gemfile" ]; then
 fi
 #
 # GEM_HOME in the container lives at /gems
-export GEM_HOME=/gems
+GEM_HOME=/gems
+export GEM_HOME
 #
 # Make sure the gems directory exists
 if [ ! -d "$GEM_HOME" ]; then
-    mkdir "$GEM_HOME"
+    mkdir -p "$GEM_HOME"
 fi
 #
 # Override $HOME to point at the volume-mounted directory. This is needed
 # because Bundle writes to a .bundle directory inside the user's home
 # directory.
-export HOME=/srv/home
+HOME=/srv
+export HOME
 #
 # Default to building; allows override to serving.
 if [ -z "$JEKYLL_ACTION" ]; then
-    export JEKYLL_ACTION="build"
+    JEKYLL_ACTION="build"
+    export JEKYLL_ACTION
```

Comment:
Why the separation of export and set?

Reply:
Sorry, I was cargo culting this! TL;DR: It can be reverted.

It's force of habit because I often set variables as the return of another command:
```bash
#!/bin/bash
export SHORT_REV="$( git rev-parse --short HEAD )"
printf "Git short revision: %s" "$SHORT_REV"
```

which should be:
```bash
#!/bin/bash
SHORT_REV="$( git rev-parse --short HEAD )"
export SHORT_REV
printf "Git short revision: %s" "$SHORT_REV"
```


See also [https://github.com/koalaman/shellcheck/wiki/SC2155].

Reply:
Ah - thanks. I wasn't seeing a warning because:

> Shellcheck does not warn about export foo=bar because bar is a literal and not a command substitution with an independent return value.

That's all fine.